server/catgenome/src/main/java/com/epam/catgenome/manager/lineage/LineageTreeManager.java

```diff
@@ -121,51 +123,33 @@ public LineageTree loadLineageTree(final long lineageTreeId) {
         final List<LineageTreeNode> referenceNodes = lineageTreeNodeDao.loadLineageTreeNodesByReference(referenceId);
-        final List<LineageTree> referenceTrees = new ArrayList<>();
-        final Map<Long, List<LineageTreeEdge>> edgesMap = new HashMap<>();
-        final Map<Long, Set<Long>> nodesMap = new HashMap<>();
-        final Set<Long> allNodeIds = new HashSet<>();
         final Set<Long> allTreeIds = referenceNodes.stream()
                 .map(LineageTreeNode::getLineageTreeId)
                 .collect(Collectors.toSet());
-        for (LineageTreeNode referenceNode: referenceNodes) {
-            final List<LineageTreeEdge> edges = lineageTreeEdgeDao.loadLineageTreeEdgesById(
-                    referenceNode.getLineageTreeNodeId());
-            //edges
-            edgesMap.put(referenceNode.getLineageTreeNodeId(), edges);
+        return lineageTreeDao.loadLineageTrees(allTreeIds);
+    }
 
-            //nodes
-            final Set<Long> nodeIds = new HashSet<>();
-            nodeIds.addAll(edges.stream().map(LineageTreeEdge::getNodeFromId).collect(Collectors.toList()));
-            nodeIds.addAll(edges.stream().map(LineageTreeEdge::getNodeToId).collect(Collectors.toList()));
-            nodeIds.add(referenceNode.getLineageTreeNodeId());
-            nodesMap.put(referenceNode.getLineageTreeNodeId(), nodeIds);
-            allNodeIds.addAll(nodeIds);
-        }
-        final List<LineageTreeNode> allNodes = lineageTreeNodeDao.loadLineageTreeNodesById(allNodeIds);
-        final List<LineageTree> allTrees = lineageTreeDao.loadLineageTrees(allTreeIds);
-        for (LineageTreeNode node: referenceNodes) {
-            final long referenceNodeId = node.getLineageTreeNodeId();
-            final Set<Long> nodeIds = nodesMap.get(referenceNodeId);
-            LineageTree tree = allTrees.stream()
-                    .filter(t -> t.getLineageTreeId().equals(node.getLineageTreeId()))
-                    .findFirst()
-                    .orElse(null);
-            List<LineageTreeNode> nodes = allNodes.stream()
-                    .filter(n -> nodeIds.contains(n.getLineageTreeNodeId()))
-                    .collect(Collectors.toList());
+    public List<LineageTree> loadAllLineageTrees() {
+        return lineageTreeDao.loadAllLineageTrees();
+    }
 
-            if (tree != null) {
-                tree.setEdges(edgesMap.get(referenceNodeId));
-                tree.setNodes(nodes);
-                referenceTrees.add(tree);
-            }
+    public static Map<String, String> parseAttributes(final String attributes) {
+        if (TextUtils.isEmpty(attributes)) {
+            return null;
         }
-        return referenceTrees;
+        final Map<String, String> map = new HashMap<>();
+        Arrays.stream(attributes.split(",")).forEach(a -> {
+            final String[] attr = a.split("=");
+            Assert.isTrue(attr.length == 2, getMessage(MessagesConstants.ERROR_INCORRECT_FILE_FORMAT));
```

Comment:
Let's add incorrect attribute to error message to make debugging easier, e.g. 
``Invalid attribute value: {}``
Same goes for format check in other methods of this class (file parsing)